pkg/serviceaccounts/controllers/docker_registry_service.go

```diff
@@ -242,16 +279,27 @@ func (e *DockerRegistryServiceController) getDockerRegistryLocations() []string
 		return []string{}
 	}
 	service := obj.(*kapi.Service)
+	result := []string{}
 
 	hasClusterIP := (len(service.Spec.ClusterIP) > 0) && (net.ParseIP(service.Spec.ClusterIP) != nil)
 	if hasClusterIP && len(service.Spec.Ports) > 0 {
-		return []string{
+		result = append(result, []string{
 			net.JoinHostPort(service.Spec.ClusterIP, fmt.Sprintf("%d", service.Spec.Ports[0].Port)),
 			net.JoinHostPort(fmt.Sprintf("%s.%s.svc", service.Name, service.Namespace), fmt.Sprintf("%d", service.Spec.Ports[0].Port)),
+		}...)
+	}
+
+	if objs := e.routeCache.List(); len(objs) > 0 {
+		route := objs[0].(*routeapi.Route)
+		// TODO: Can a router have ports other than 80?
```

Comment:
You could iterate over all of the routers and call the below on those matching route according to doc: https://docs.openshift.org/latest/install_config/router/default_haproxy_router.html#using-router-shards

I presume we already have some helper for router matching based on route labels.